### DISCUSSION TITLE
Requirement of \r\n Line Endings for Values in SIP Messages by HeaderScanner's scanChunk Function

### DISCUSSION THREAD

**Discussion by ChuLiqiang** [resiprocate/resiprocate#359] [Q&A] [answered]
Why does the scanChunk function in HeaderScanner require that the values in the SIP message must end with \r\n, and that just ending with \n is considered an error?
"User-Agent: DecodeDev" ends with just \n instead of \r\n. scanChunk directly detects this as an error。
The "User-Agent: DecodeDev" header value in a SIP message ends with just \n instead of \r\n. However, all the other header values in the message end with the expected \r\n line endings. The scanChunk function in HeaderScanner detects the line ending of just \n for the "User-Agent" value as an error, despite the other header value line endings being correct.
Does this accurately capture the situation you were describing? Please let me know if I should modify or expand my understanding further.

![image](https://github.com/resiprocate/resiprocate/assets/2747722/07c591a1-2c7e-46c9-bab6-8a573202a5e6)

```
REGISTER sip:71000000002000000110@192.168.1.111:5061 SIP/2.0
Via: SIP/2.0/UDP 192.168.1.114:5060;rport;branch=z9hG4bK427831878
From: <sip:42010101041140000001@7100000000>;tag=1550995956
To: <sip:42010101041140000001@7100000000>
Call-ID: 1508865972
CSeq: 1 REGISTER
Contact: <sip:42010101041140000001@192.168.1.114:5060>
Max-Forwards: 70
User-Agent: DecodeDev
X-GB-Ver:3.0
Expires: 3600
Content-Length: 0
```

**Comment by sgodin** [comment, marked as answer]
RFC3261 ABNF grammar specifies that CRLF is required for a line ending:

Request        =  Request-Line
                  *( message-header )
                  CRLF
                  [ message-body ]
....
message-header  =  (Accept
                /  Accept-Encoding
                /  Accept-Language
....
                /  WWW-Authenticate
                /  extension-header) CRLF
